### PR TITLE
Heatmap: Add datalink support

### PR DESCRIPTION
**Pull request by kureshimiru**
Enables showing of datalinks in the tooltip when hovering over heatmap cells. Datalinks for heatmaps were previously left undefined.

### CODE REVIEW

File: public/app/plugins/panel/heatmap/HeatmapHoverView.tsx [hunk 1]
```diff
@@ -120,7 +125,14 @@ const HeatmapHoverCell = ({ data, hover, showHistogram }: Props) => {
...
 
   for (const field of visibleFields ?? []) {
-    // TODO: Currently always undefined? (getLinks)
+    const hasLinks = field.config.links && field.config.links.length > 0;
+    if (hasLinks && data.heatmap) {
+      let appropriateScopedVars = scopedVars.filter(
+        (sv) => sv && sv.__dataContext && sv.__dataContext.value.field.name === nonNumericOrdinalDisplay
+      )[0];
+      field.getLinks = getLinksSupplier(data.heatmap, field, appropriateScopedVars ?? {}, replaceVars);
+    }
```
Comment (nmarrs) [inline public/app/plugins/panel/heatmap/HeatmapHoverView.tsx:134]: nit - but maybe something more along the lines of

```suggestion
const hasLinks = field.config.links && field.config.links.length > 0;

if (hasLinks && data.heatmap) {
  const appropriateScopedVars = scopedVars.find((scopedVar) => scopedVar && scopedVar.__dataContext && scopedVar.__dataContext.value.field.name === nonNumericOrdinalDisplay);

  field.getLinks = getLinksSupplier(data.heatmap, field, appropriateScopedVars || {}, replaceVars);
}
```

File: public/app/plugins/panel/heatmap/HeatmapPanel.tsx [hunk 2]
```diff
@@ -43,6 +43,16 @@ export const HeatmapPanel = ({
   const styles = useStyles2(getStyles);
   const { sync } = usePanelContext();
 
+  //  necessary for enabling datalinks in hover view
+  let scopedVarsFromRawData = [];
+  for (const series of data.series) {
+    for (const field of series.fields) {
+      if (field.state?.scopedVars) {
+        scopedVarsFromRawData.push(field.state?.scopedVars);
```
Comment (nmarrs) [inline public/app/plugins/panel/heatmap/HeatmapPanel.tsx:51]: another nit
```suggestion
        scopedVarsFromRawData.push(field.state.scopedVars);
```

**Review by nmarrs — COMMENTED** [review]
A few code cleanup items :)

**Review by nmarrs — APPROVED** [review]
We want to land this for 10.1 so I'm going to approve and merge. I will push up a subsequent PR for the clean up items. I'm unable to push changes to your repo, but don't want to create a new PR as this is a great community contribution and you deserve credit :)

**Comment by kureshimiru** [comment]
@grafana

> We want to land this for 10.1 so I'm going to approve and merge. I will push up a subsequent PR for the clean up items. I'm unable to push changes to your repo, but don't want to create a new PR as this is a great community contribution and you deserve credit :)

Thank you very much! :)
I'm sorry for non-responsiveness, we are in the middle of a PI planning event so yesterday and today are pretty busy.

**Pull request merged**